lemminx-maven/src/main/java/org/eclipse/lemminx/maven/MavenCompletionParticipant.java

```diff
@@ -105,12 +105,12 @@ public NodeWithChildrenInsertionStrategy(String elementName) {
 	private final MavenSession mavenSession;
 	private final BuildPluginManager buildPluginManager;
 
-	public MavenCompletionParticipant(MavenProjectCache cache, LocalRepositorySearcher localRepositorySearcher, RemoteRepositoryIndexSearcher indexSearcher, RepositorySystemSession repoSession, MavenSession mavenSession, MavenPluginManager pluginManager, BuildPluginManager buildPluginManager) {
+	public MavenCompletionParticipant(MavenProjectCache cache, LocalRepositorySearcher localRepositorySearcher, RemoteRepositoryIndexSearcher indexSearcher, MavenSession mavenSession, MavenPluginManager pluginManager, BuildPluginManager buildPluginManager) {
```

Comment:
@mickaelistria this didn't shorten the constructors as much as I had hoped. What do you think about making a container class to hold things such as the MavenProjectCache, LocalRepositorySearcher,  RemoteRepositoryIndexSearcher, MavenSession etc.

Reply:
IMO, it's not worth the effort. A long constructor list isn't a too bad thing per se as it doesn't cause bugs, doesn't have a performance cost...; so introducing more code and more classes without clear design gain is not really helping.

Reply:
That said, the current change is very good as it removes some useless args we can easily deduce from other ones.

Reply:
Okay sounds good with me. Will merge.